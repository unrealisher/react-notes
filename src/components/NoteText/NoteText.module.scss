.wrapper {
  height: 100%;
  box-sizing: border-box;
  position: relative;
  z-index: 2;
  display: flex;
  flex-direction: column;
  padding: 24px;
  border-radius: 16px;
}

.wrapper_down {
  display: flex;
  flex-direction: column;
  margin-top: auto;
}

.title {
  font-size: 24px;
  line-height: 28px;
  margin-bottom: 12px;
}

.text {
  margin-bottom: 12px;
}
