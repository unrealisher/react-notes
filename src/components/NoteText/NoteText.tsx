import React from "react";

import INote from "../../interfaces/INote";

import NoteFooter from "./../NoteFooter/NoteFooter";

import styles from "./NoteText.module.scss";

interface IProps {
  note?: INote;
  color?: string;
  tags?: string[] | undefined;
  setPatchItem: Function;
  setPopup: Function;
}

export const NoteText = (props: IProps): JSX.Element => {
  const { note, color, tags, setPatchItem, setPopup } = props;

  return (
    <React.Fragment>
      <div className={styles.wrapper}>
        <div
          className={styles.wrapper_color}
          style={{ backgroundColor: color + "66" }}
        >
          {note && note.title !== undefined && (
            <h3 className={styles.title}>{note.title}</h3>
          )}
          {note && note.text !== undefined && (
            <p className={styles.text}>{note.text}</p>
          )}
          {note && (
            <NoteFooter
              tags={tags}
              note={note}
              setPatchItem={setPatchItem}
              setPopup={setPopup}
            />
          )}
        </div>
      </div>
    </React.Fragment>
  );
};

export default NoteText;
