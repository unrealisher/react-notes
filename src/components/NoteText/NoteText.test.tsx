import React from "react";
import { shallow } from "enzyme";
import { NoteText } from "./NoteText";

describe("NoteText", () => {
  const note = {
    id: 1,
    type: "text",
    title: "Завтра дедлайн",
    text: "Не забыть сверстать макет",
    tags: [1, 2],
    color: 2,
    size: "s",
    created: 1551714600000
  };
  const color = "#F2C94C";
  const tags = ["Покупки", "Работа", "ШРИ"];
  const setPatchItem = (): void => {};
  const setPopup = (): void => {};
  it("correct render", () => {
    const component = shallow(
      <NoteText
        note={note}
        color={color}
        tags={tags}
        setPatchItem={setPatchItem}
        setPopup={setPopup}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
