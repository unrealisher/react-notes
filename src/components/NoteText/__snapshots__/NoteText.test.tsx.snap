// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteText correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteText
    color="#F2C94C"
    note={
      Object {
        "color": 2,
        "created": 1551714600000,
        "id": 1,
        "size": "s",
        "tags": Array [
          1,
          2,
        ],
        "text": "Не забыть сверстать макет",
        "title": "Завтра дедлайн",
        "type": "text",
      }
    }
    setPatchItem={[Function]}
    setPopup={[Function]}
    tags={
      Array [
        "Покупки",
        "Работа",
        "ШРИ",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="wrapper"
      >
        <div
          className="wrapper_color"
          style={
            Object {
              "backgroundColor": "#F2C94C66",
            }
          }
        >
          <h3
            className="title"
          >
            Завтра дедлайн
          </h3>
          <p
            className="text"
          >
            Не забыть сверстать макет
          </p>
          <Connect(NoteFooter)
            note={
              Object {
                "color": 2,
                "created": 1551714600000,
                "id": 1,
                "size": "s",
                "tags": Array [
                  1,
                  2,
                ],
                "text": "Не забыть сверстать макет",
                "title": "Завтра дедлайн",
                "type": "text",
              }
            }
            setPatchItem={[Function]}
            setPopup={[Function]}
            tags={
              Array [
                "Покупки",
                "Работа",
                "ШРИ",
              ]
            }
          />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="wrapper_color"
          style={
            Object {
              "backgroundColor": "#F2C94C66",
            }
          }
        >
          <h3
            className="title"
          >
            Завтра дедлайн
          </h3>
          <p
            className="text"
          >
            Не забыть сверстать макет
          </p>
          <Connect(NoteFooter)
            note={
              Object {
                "color": 2,
                "created": 1551714600000,
                "id": 1,
                "size": "s",
                "tags": Array [
                  1,
                  2,
                ],
                "text": "Не забыть сверстать макет",
                "title": "Завтра дедлайн",
                "type": "text",
              }
            }
            setPatchItem={[Function]}
            setPopup={[Function]}
            tags={
              Array [
                "Покупки",
                "Работа",
                "ШРИ",
              ]
            }
          />
        </div>,
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="title"
            >
              Завтра дедлайн
            </h3>,
            <p
              className="text"
            >
              Не забыть сверстать макет
            </p>,
            <Connect(NoteFooter)
              note={
                Object {
                  "color": 2,
                  "created": 1551714600000,
                  "id": 1,
                  "size": "s",
                  "tags": Array [
                    1,
                    2,
                  ],
                  "text": "Не забыть сверстать макет",
                  "title": "Завтра дедлайн",
                  "type": "text",
                }
              }
              setPatchItem={[Function]}
              setPopup={[Function]}
              tags={
                Array [
                  "Покупки",
                  "Работа",
                  "ШРИ",
                ]
              }
            />,
          ],
          "className": "wrapper_color",
          "style": Object {
            "backgroundColor": "#F2C94C66",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Завтра дедлайн",
              "className": "title",
            },
            "ref": null,
            "rendered": "Завтра дедлайн",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Не забыть сверстать макет",
              "className": "text",
            },
            "ref": null,
            "rendered": "Не забыть сверстать макет",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "note": Object {
                "color": 2,
                "created": 1551714600000,
                "id": 1,
                "size": "s",
                "tags": Array [
                  1,
                  2,
                ],
                "text": "Не забыть сверстать макет",
                "title": "Завтра дедлайн",
                "type": "text",
              },
              "setPatchItem": [Function],
              "setPopup": [Function],
              "tags": Array [
                "Покупки",
                "Работа",
                "ШРИ",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="wrapper"
        >
          <div
            className="wrapper_color"
            style={
              Object {
                "backgroundColor": "#F2C94C66",
              }
            }
          >
            <h3
              className="title"
            >
              Завтра дедлайн
            </h3>
            <p
              className="text"
            >
              Не забыть сверстать макет
            </p>
            <Connect(NoteFooter)
              note={
                Object {
                  "color": 2,
                  "created": 1551714600000,
                  "id": 1,
                  "size": "s",
                  "tags": Array [
                    1,
                    2,
                  ],
                  "text": "Не забыть сверстать макет",
                  "title": "Завтра дедлайн",
                  "type": "text",
                }
              }
              setPatchItem={[Function]}
              setPopup={[Function]}
              tags={
                Array [
                  "Покупки",
                  "Работа",
                  "ШРИ",
                ]
              }
            />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="wrapper_color"
            style={
              Object {
                "backgroundColor": "#F2C94C66",
              }
            }
          >
            <h3
              className="title"
            >
              Завтра дедлайн
            </h3>
            <p
              className="text"
            >
              Не забыть сверстать макет
            </p>
            <Connect(NoteFooter)
              note={
                Object {
                  "color": 2,
                  "created": 1551714600000,
                  "id": 1,
                  "size": "s",
                  "tags": Array [
                    1,
                    2,
                  ],
                  "text": "Не забыть сверстать макет",
                  "title": "Завтра дедлайн",
                  "type": "text",
                }
              }
              setPatchItem={[Function]}
              setPopup={[Function]}
              tags={
                Array [
                  "Покупки",
                  "Работа",
                  "ШРИ",
                ]
              }
            />
          </div>,
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="title"
              >
                Завтра дедлайн
              </h3>,
              <p
                className="text"
              >
                Не забыть сверстать макет
              </p>,
              <Connect(NoteFooter)
                note={
                  Object {
                    "color": 2,
                    "created": 1551714600000,
                    "id": 1,
                    "size": "s",
                    "tags": Array [
                      1,
                      2,
                    ],
                    "text": "Не забыть сверстать макет",
                    "title": "Завтра дедлайн",
                    "type": "text",
                  }
                }
                setPatchItem={[Function]}
                setPopup={[Function]}
                tags={
                  Array [
                    "Покупки",
                    "Работа",
                    "ШРИ",
                  ]
                }
              />,
            ],
            "className": "wrapper_color",
            "style": Object {
              "backgroundColor": "#F2C94C66",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Завтра дедлайн",
                "className": "title",
              },
              "ref": null,
              "rendered": "Завтра дедлайн",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Не забыть сверстать макет",
                "className": "text",
              },
              "ref": null,
              "rendered": "Не забыть сверстать макет",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "note": Object {
                  "color": 2,
                  "created": 1551714600000,
                  "id": 1,
                  "size": "s",
                  "tags": Array [
                    1,
                    2,
                  ],
                  "text": "Не забыть сверстать макет",
                  "title": "Завтра дедлайн",
                  "type": "text",
                },
                "setPatchItem": [Function],
                "setPopup": [Function],
                "tags": Array [
                  "Покупки",
                  "Работа",
                  "ШРИ",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
